src/server/acl/acl_family.cc

```diff
@@ -0,0 +1,108 @@
+// Copyright 2022, DragonflyDB authors.  All rights reserved.
+// See LICENSE for licensing terms.
+
+#include "server/acl/acl_family.h"
+
+#include "server/acl/acl_commands_def.h"
+#include "server/command_registry.h"
+#include "server/conn_context.h"
+
+namespace dfly::acl {
+
+constexpr uint32_t kList = acl::ADMIN | acl::SLOW | acl::DANGEROUS;
+
+static std::string AclToString(uint32_t acl_category) {
+  std::string tmp;
+
+  if (acl_category == acl::ALL) {
+    return "+@all";
+  }
+
+  if (acl_category == acl::NONE) {
+    return "+@none";
+  }
+
+  const std::string prefix = "+@";
+  const std::string postfix = " ";
+
+  for (uint32_t step = 0, cat = 0; cat != JSON; cat = 1ULL << ++step) {
+    if (acl_category & cat) {
+      tmp += prefix;
+      tmp += REVERSE_CATEGORY_INDEX_TABLE[step];
+      tmp += postfix;
+    }
+  }
+  tmp.erase(tmp.size());
+
+  return tmp;
+}
+
+void AclFamily::List(CmdArgList args, ConnectionContext* cntx) {
+  auto registry_with_lock = cntx->user_registry->GetRegistryWithLock();
+  const auto& registry = registry_with_lock.registry;
+
+  std::string buffer(255, ' ');
```

Comment:
I would be surprised if any line surpasses the size of this buffer. But even if it does, the resize function will take care of it. We basically get away with 1 alloc most of the time...